src/App.tsx

```diff
@@ -15,36 +15,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  None = 'none',
+  Alphabetically = 'alphabetically',
+  ByLength = 'length',
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const [sortMethod, setSortMethod] = useState<SortType>(SortType.None);
+
+  const sortGoods = (method: SortType, reverse: boolean): string[] => {
+    const sortedGoods = [...goodsFromServer];
+
+    if (method === SortType.Alphabetically) {
+      sortedGoods.sort();
+    } else if (method === SortType.ByLength) {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (reverse) {
+      sortedGoods.reverse();
+    }
+
+    return sortedGoods;
+  };
```

Comment:
[CODE KNOWLEDGE] - The sortGoods function is being called in each render which can lead to unnecessary computations. Consider using useMemo hook to optimize this. The useMemo will recompute the memoized value only when one of the dependencies has changed.